sql/core/src/test/resources/sql-tests/inputs/misc-functions.sql

```diff
@@ -0,0 +1,10 @@
+-- test for misc functions
+
+-- typeof
+select typeof(null);
+select typeof(true);
+select typeof(1Y), typeof(1S), typeof(1), typeof(1L);
```

Comment:
do you know why these tests pass before we add `UTF8String.fromString`?

Reply:
ah I got it. The rule constant folding calls `Literal.create(e.eval(EmptyRow), e.dataType)`, which will converts the data to internal format.

Now I'm curious about how we expose this bug with UDT...

Reply:
In `CatalystTypeConverters`
```scala
 /**
   *  Converts Scala objects to Catalyst rows / types.
   *
   *  Note: This should be called before do evaluation on Row
   *        (It does not support UDT)
   *  This is used to create an RDD or test results with correct types for Catalyst.
   */
  def convertToCatalyst(a: Any): Any = a match {
```